mlir/tools/rocmlir-gen/rocmlir-gen.cpp

```diff
@@ -2133,87 +2142,65 @@ static func::FuncOp createVerifierFunc(ModuleOp module, const KernelIF &kernel,
   auto printDebugVal =
       b.create<arith::ConstantIntOp>(loc, printDebug, charType);
 
-  // obtain function name of the verifier wrapper
-  std::string verifyFuncName = "mcpuVerify";
-  if (valElemType.isF32()) {
-    verifyFuncName += "Float";
-  } else if (valElemType.isInteger(32)) {
-    verifyFuncName += "Int32";
-  } else if (valElemType.isInteger(64)) {
-    verifyFuncName += "Int32Int64";
-  } else if (valElemType.isInteger(8)) {
-    verifyFuncName += "Int8";
-  } else {
-    llvm::errs() << "Unsupported type of validation function output: ";
-    llvm::errs() << " (Only f32, int32 and int64 are supported)\n";
-    exit(1);
-  }
+  Value testResult, valResult;       // Values passed to the verify function
+  Value testResultNew, valResultNew; // Values used for type conversion
 
   auto mr1DUnkTestType =
       MemRefType::get({mlir::ShapedType::kDynamic}, testOutType);
   auto mr1DUnkValType =
       MemRefType::get({mlir::ShapedType::kDynamic}, valElemType);
+  auto mr1DUnkF32Type =
+      MemRefType::get({mlir::ShapedType::kDynamic}, floatType);
 
-  bool isTestAndValSameType =
-      (testOutType.isInteger(32) || testOutType.isF32());
-
-  Value testResult, valResult; // Values passed to the verify function
-  Value testResultNew;         // Values used for type conversion
-  if (!isTestAndValSameType) {
+  if (testOutType.isa<FloatType>() && !testOutType.isF32()) {
```

Comment:
The old side, using the isTestAndValSameType variable (not quite a descriptive name), would convert the f16 kernel results to f32, and take the validation results (which were always f32) and convert to f16 and back to mimic the same precision.

The new side converts both kernel and validation results identically, since they're going to be the same type -- if float and not f32, convert to f32.

Both sides then pass the results to the mcpuVerify function.